queries/bash/highlights.scm

```diff
@@ -0,0 +1,110 @@
+[
+ "("
+ ")"
+ "{"
+ "}"
+ "["
+ "]"
+ ] @punctuation.bracket
+
+[
+ ";"
+ ";;"
+ (heredoc_start)
+ ] @punctuation.delimiter
+
+[
+ ">"
+ "<"
+ "&"
+ "&&"
+ "|"
+ "||"
+ "="
+ "=="
+ "!="
+ ] @operator
+
+[
+ (string) 
+ (raw_string)
+ (heredoc_body)
+] @string
+
+[
+ "if"
+ "then"
+ "else"
+ "elif"
+ "fi"
+ "case"
+ "in"
+ "esac"
+ ] @conditional
+
+[
+ "for"
+ "do"
+ "done"
+ "while"
+ ] @repeat
+
+[
+ "declare"
+ "export"
+ "local"
+ "readonly"
+ "unset"
+ ] @keyword
+
+[
+ (special_variable_name) 
+ ("$" (special_variable_name))
+ ] @constant
+
+((word) @constant
+  (#match? @constant "SIG(INT|TERM|QUIT|TIN|TOU|STP|HUP)"))
+
+((word) @boolean
+  (#match? @boolean "true|false"))
+
+((word) @number
+  (#match? @number "^\d*$"))
+
+(comment) @comment
+(test_operator) @string.
+
+(command_substitution
+  [ "$(" ")" ] @punctuation.bracket)
+
+
+(function_definition
+  name: (word) @function)
+
+(command_name (word)) @function
+
+(command 
+  argument: [
+             (word) @parameter
+             ((word) @number
+              (#match? @number "^\d*$"))
+             (concatenation (word) @parameter)
+             ])
```

Comment:
Wow that's a tricky one ! Isn't the `@number` case already covered ?
And the `@parameter` seem to be convered below too

Reply:
Number has to be explicit here because "(word)" has already been set to "@parameter". The more specific the query the more precidence it seems to have. so if you override in a specific query like this one. you will need to specify again if you have predicates. The difficult thing about the bash parser is that I had regex out the numbers myself :/

Reply:
But that seems to work so its okay